### ISSUE TITLE
ZFS mount point configuration

### ISSUE THREAD

**Issue by immens** [openebs/zfs-localpv#498]
**Describe the problem/challenge you have**
For bare metal installations (e.g. home server) with only 1 node it will be great to have ability to mount datasets automatically (data access, backups, etc)

**Describe the solution you'd like**
```❯ zfs list
NAME                                                       USED  AVAIL  REFER  MOUNTPOINT
zfast                                                     72.7M   231G    96K  none
zfast/data                                                15.5M   231G  15.4M  /storage/fast
zfast/data/k8s                                              96K   231G    96K  /storage/fast/k8s
zfast/data/k8s/pvc-19aeff59-85b2-46de-9431-d3ec85fb12ca    100K  5.00G   100K  legacy
```

In setups like this one, PVC's dataset can be automatically mounted e.g. `<parent's mount point>/pvc-<PVC's name>-<some hash or uuid>`. Maybe somehow correlating with `shared` prop

**Environment:**
- ZFS-LocalPV version: 2.4.0 (helm chart 3.10.0)
- Kubernetes version: v1.29.1
- Kubernetes installer & version: `kubeadm` v1.29.1
- Cloud provider or hardware configuration: none
- OS (e.g. from `/etc/os-release`): Arch Linux (rolling); 6.6.12-1-lts, ZFS 2.2.2

**Comment by hrudaya21** [comment]
@immens Currently it is not supported in the product. Team is handsfull with other openEBS engine. It would be great, if you can contribute for this feature to the community.

**Comment by avishnu** [comment]
Where and when are are expecting the datasets to be mounted automatically? It's not clear from the description.

**Comment by Abhinandan-Purkait** [comment]
@immens Can you answer the above question for more clarity on his?

**Comment by immens** [comment]
Sorry for the late response.

The idea was to provide the ability to mount the created volumes to a directory specified as the mount point of the parent dataset (defined in the StorageClass), or perhaps to allow specifying a base path for mounting directly in the StorageClass properties.

In the example from the issue, `zfast/data/k8s` is the dataset specified in the StorageClass properties, and it has the mount point set to `/storage/fast/k8s`.

```yaml
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  name: zfs-fast
parameters:
  fstype: "zfs"
  poolname: "zfast/data/k8s"
provisioner: zfs.csi.openebs.io
volumeBindingMode: WaitForFirstConsumer
```

**Comment by Abhinandan-Purkait** [comment]
@immens Do you mean to say a path to be provided on the storage class parameter that mounts the volume on creation to that path even without volume being published anywhere? Or is it like an extra mount point beside the csi staging path?

**Comment by skirsten** [comment]
I would also really like this because it would allow to enable the `sharenfs` feature. Currently this is not working because it does not work with legacy mountpoint.

**Comment by immens** [comment]
> [@immens](https://github.com/immens) Do you mean to say a path to be provided on the storage class parameter that mounts the volume on creation to that path even without volume being published anywhere? Or is it like an extra mount point beside the csi staging path?

Yes — an optional parameter in the StorageClass (e.g., hostMountPath) that would make the ZFS dataset automatically mount at the specified path on the node as soon as the volume is created, independent of whether it’s published to a Pod. This would solve the need for direct host‑side access (backups, sharenfs, etc.).
